src/graphiql.js

```diff
@@ -0,0 +1,118 @@
+import React, { useRef, useEffect, useState } from 'react';
+import GraphiQL from 'graphiql';
+import PropTypes from 'prop-types';
+import { buildSchema } from 'graphql';
+import GraphiQLExplorer from 'graphiql-explorer';
+import './common/styles/index.scss';
+
+const TykGraphiQL = (props) => {
+  const {
+    fetcher,
+    schema,
+    ideName = 'TykGraphiQL',
+    query: initialQuery = '{}',
+    showPlayground = true,
+    menu,
+  } = props;
+  const playgroundWrapperRef = useRef(null);
+  let playgroundRef = useRef(null);
+  const [explorerIsOpen, setExplorerIsOpen] = useState(false);
+  const [query, setQuery] = useState(initialQuery);
+  const [selectedMenu, setSelectedMenu] = useState(menu?.title);
+
+  useEffect(() => {
+    if (!playgroundWrapperRef.current) {
+      return;
+    }
+
+    playgroundWrapperRef.current.addEventListener('click', (e) => {
+      e.preventDefault();
+    });
+  }, [playgroundWrapperRef]);
+
+  const toggleExplorer = () => setExplorerIsOpen(!explorerIsOpen);
+
+  const handleMenuSelection = (item) => {
+    menu.onSelect(item);
+    setSelectedMenu(item?.label);
+  };
+
+  return (
+    <div
+      ref={playgroundWrapperRef}
+      className="graphiql-container"
+      style={{
+        visibility: showPlayground ? 'visible' : 'hidden',
+      }}
+    >
+      <GraphiQLExplorer
+        explorerIsOpen={explorerIsOpen}
+        onToggleExplorer={toggleExplorer}
+        onEdit={(q) => setQuery(q)}
```

Comment:
In general `x => f(x)` is the same thing as `f`.
So in this case `onEdit={(q) => setQuery(q)}` can be `onEdit={setQuery}`.